packages/aws-cdk/lib/api/deployments.ts

```diff
@@ -419,27 +419,26 @@ export class Deployments {
     const { stackSdk, resolvedEnvironment, envResources } = await this.prepareSdkFor(stackArtifact, undefined, Mode.ForReading);
     const cfn = stackSdk.cloudFormation();
 
-    await uploadStackTemplateAssets(stackArtifact, this);
+    const proof = await uploadStackTemplateAssets(stackArtifact, this.sdkProvider, resolvedEnvironment);
```

Comment:
Unrelated, but why does this method even publish assets?

Reply:
Every API that calls CloudFormation with a template may need to publish assets, because with a `LegacyStackSynthesizer` the stack template is not automatically written to disk and published; the CLI will need to do that just-in-time.